src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -0,0 +1,172 @@
+end
+
+"""
+    terminal_velocity(q_rai, ρ)
+
+where:
+  - `q_rai` - rain water specific humidity
+  - `ρ`     - density of air
+
+Returns the mass weighted average rain terminal velocity assuming
+Marshall Palmer 1948 distribution of rain drops.
+"""
+function terminal_velocity(q_rai::DT, ρ::DT) where {DT<:Real}
+
+    v_c = terminal_velocity_single_drop_coeff(ρ)
+
+    # gamma(9/2)
+    gamma_9_2 = DT(11.631728396567448)
+
+    lambda::DT = (DT(8) * π * dens_liquid * MP_n_0 / ρ / q_rai)^DT(1/4)
+
+    return gamma_9_2 * v_c / DT(6) * sqrt(grav / lambda)
+end
+
+
+"""
+    conv_q_vap_to_q_liq(q_sat, q)
+
+where:
+- `q_sat` - PhasePartition at equilibrium
+- `q`     - current PhasePartition
+
+Returns the q_liq tendency due to condensation/evaporation.
+The tendency is obtained assuming a relaxation to equilibrium with
+constant timescale.
+"""
+function conv_q_vap_to_q_liq(q_sat::PhasePartition{DT},
+                             q::PhasePartition{DT}) where {DT<:Real}
+
+  if q_sat.ice != DT(0)
+    error("1-moment bulk microphysics is not defined for snow/ice")
+    #This should be the q_ice tendency due to sublimation/resublimation.
+    #src_q_ice = (q_sat.ice - q.ice) / τ_subl_resubl
+  end
+
+  return (q_sat.liq - q.liq) / τ_cond_evap
+end
+
+
+"""
+    conv_q_liq_to_q_rai_acnv(q_liq)
+
+where:
+- `q_liq` - is the liquid water specific humidity
+
+Returns the q_rai tendency due to collisions between cloud droplets
+(autoconversion) parametrized following Kessler 1995.
+"""
+function conv_q_liq_to_q_rai_acnv(q_liq::DT) where {DT<:Real}
+
+  return max(DT(0), q_liq - q_liq_threshold) / τ_acnv
+end
+
+
+"""
+    conv_q_liq_to_q_rai_accr(q_liq, q_rai, ρ)
+
+where:
+- `q_liq` - is the liquid water specific humidity
+- `q_rai` - is the rain water specific humidity
+- `ρ` - is the density of air
+
+Returns the q_rai tendency due to collisions between cloud droplets
+and rain drops (accretion) parametrized following Kessler 1995.
+"""
+function conv_q_liq_to_q_rai_accr(q_liq::DT, q_rai::DT, ρ::DT) where {DT<:Real}
+
+  # terminal_vel_of_individual_drop = v_drop_coeff * drop_radius^(1/2)
+  v_c::DT = terminal_velocity_single_drop_coeff(ρ)
+
+  #gamma(7/2)
+  gamma_7_2 = DT(3.3233509704478426)
+
+  accr_coeff::DT = gamma_7_2 * DT(8)^DT(-7/8) * π^DT(1/8) * v_c * E_col *
+                   (ρ / dens_liquid)^DT(7/8)
+
+  return accr_coeff * DT(MP_n_0)^DT(1/8) * sqrt(DT(grav)) *
+         q_liq * q_rai^DT(7/8)
+end
+
+"""
+    conv_q_rai_to_q_vap(q_rai, q, T, p, ρ)
+
+where:
+ - q_rai - rain water specific humidity
+ - q - current PhasePartition
+ - T - temperature
+ - p - pressure
+ - ρ - air density
+
+Returns the q_rai tendency due to rain evaporation. Parameterized following
+Smolarkiewicz and Grabowski 1996.
+"""
+function conv_q_rai_to_q_vap(qr::DT, q::PhasePartition{DT},
+                             T::DT, p::DT, ρ::DT) where {DT<:Real}
+
+  qv_sat = saturation_shum(T, ρ, q)
+  q_v::DT = q.tot - q.liq - q.ice
```

Comment:
```suggestion
  q_v = q.tot - q.liq - q.ice
```
The type is already inherited from the passed in argument

Reply:
done